### PR TITLE
Model error visualization fix

### LINKED ISSUE #1021: [BUG] Issues with Model Error Analysis display and conditions
**Describe the bug**
Bugs raised by @shir22 - 
1. Error values are very high, needs to be checked
2. Condition says that decrease in performance is 8%, while the decrease in performance from accuracy 1 to accuracy 0.75 is larger. 
Apart from that - the units (and nature) of the Error y axis is not clear and should be elaborated in the display.

**To Reproduce**
See Model Error Analysis results in the iris example https://docs.deepchecks.com/en/stable/examples/guides/quickstart_in_5_minutes.html

**Screenshots**
image
image

### PR DESCRIPTION
**Pull request by JKL98ISR**
close #1021

### CODE REVIEW

File: deepchecks/utils/single_sample_metrics.py [hunk 1]
```diff
@@ -10,13 +10,60 @@
...
+
+    # Make y_true into one-hot
+    lb = LabelBinarizer()
+    lb.fit(list(range(y_pred.shape[1])))  # We assume that y_true is a list of corresponding indices in y_pred
+    transformed_labels = lb.transform(y_true)
+
+    if transformed_labels.shape[1] == 1:
+        transformed_labels = np.append(
+            1 - transformed_labels, transformed_labels, axis=1
+        )
```
Comment (nirhutnik) [inline deepchecks/utils/single_sample_metrics.py:60]: Remove if block

File: deepchecks/utils/single_sample_metrics.py [hunk 2]
```diff
@@ -10,13 +10,60 @@
...
+    OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    """
     y_true = np.array(y_true)
-    return - (np.tile(y_true.reshape((-1, 1)), (1, y_pred.shape[1])) *
-              np.log(y_pred + np.finfo(float).eps)).sum(axis=1)
+
+    # Make y_true into one-hot
+    lb = LabelBinarizer()
+    lb.fit(list(range(y_pred.shape[1])))  # We assume that y_true is a list of corresponding indices in y_pred
```
Comment (nirhutnik) [inline deepchecks/utils/single_sample_metrics.py:54]: Fix comment - make it clearer ("we assume both y_pred and y_true correlate to the same samples" or something)

File: docs/source/user-guide/general/exporting_results/plot_export_outputs_to_json.py [hunk 3]
```diff
@@ -49,7 +49,7 @@
 # Serialization of the output to a JSON format is possible using the ``to_json`` function.
 # This function takes the check outputs and serializes it to a JSON string.
 #
-# The format of the check's output json includes all of the info required to reconstruct
+# The format of the check's output json includes all the info required to reconstruct
```
Comment (nirhutnik) [inline docs/source/user-guide/general/exporting_results/plot_export_outputs_to_json.py:52]:
```suggestion
# The format of the check's output json includes all info required to reconstruct
```

File: deepchecks/utils/single_sample_metrics.py [hunk 4]
```diff
@@ -10,13 +10,60 @@
...
+    transformed_labels = lb.transform(y_true)
+
+    if transformed_labels.shape[1] == 1:
+        transformed_labels = np.append(
+            1 - transformed_labels, transformed_labels, axis=1
+        )
+
+    # clip and renormalization y_pred
+    y_pred = np.clip(y_pred, eps, 1 - eps)
```
Comment (nirhutnik) [inline deepchecks/utils/single_sample_metrics.py:63]:
```suggestion
    y_pred = y_pred.clip(eps, 1 - eps)
```
Just because prettier

**Review by nirhutnik — APPROVED** [review]

File: deepchecks/utils/single_sample_metrics.py [hunk 5]
```diff
@@ -10,13 +10,60 @@
...
+    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+    FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+    DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+    SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+    OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    """
```
[the remark this answers is not in the document]
Comment (noamzbr) [reply deepchecks/utils/single_sample_metrics.py:49]: Don't know, but wanted to put it together with the appropriate section of code.

File: docs/source/user-guide/general/exporting_results/plot_export_outputs_to_json.py [hunk 6]
```diff
@@ -30,10 +30,10 @@
 # Run a Check
 # ===========
 
-from deepchecks.tabular.checks import ModelErrorAnalysis
+from deepchecks.tabular.checks import WholeDatasetDrift
 
-result = ModelErrorAnalysis().add_condition_segments_performance_relative_difference_not_greater_than(\
-                                ).run(train_dataset, test_dataset, model)
+result = WholeDatasetDrift().add_condition_overall_drift_value_not_greater_than(
+).run(train_dataset, test_dataset, model)
```
[the remark this answers is not in the document]
Comment (noamzbr) [reply docs/source/user-guide/general/exporting_results/plot_export_outputs_to_json.py:36]: ModelErrorAnalysis does not find a significant segmentation (and raises an error)
